packages/v1-sdk/src/entities/pair.ts

```diff
@@ -158,76 +184,121 @@ export class Pair {
 
   public getOutputAmount(inputAmount: CurrencyAmount<Token>): [CurrencyAmount<Token>, Pair] {
     invariant(this.involvesToken(inputAmount.currency), 'TOKEN');
-    if (this.reserve0.quotient === ZERO || this.reserve1.quotient === ZERO) {
-      throw new InsufficientReservesError();
-    }
 
-    const inputReserve = this.reserveOf(inputAmount.currency);
-    const outputReserve = this.reserveOf(inputAmount.currency.equals(this.token0) ? this.token1 : this.token0);
+    if (this.isStabilityFund) {
+      // this.token1 is always bnUSD
+      if (inputAmount.currency.symbol === 'bnUSD') {
+        // bnUSD -> USDC
+        // apply fee 0.1%
+        const outputAmount = CurrencyAmount.fromCurrencyAmount(inputAmount, this.token0).multiply(
```

Comment:
bnUSD -> USDC

USDC amount = bnUSD amount * (USDC/bnUSD price)
USDC/bnUSD price = 1 in this case.
use Price data structure. 

const price = new Price(this.token1, this.token0, 10n ** this.token1.decimals ., 10n ** this.token0.decimals)
const outputAmount = price.quote(inputAmount).multiply().

and remove the CurrencyAmount.fromCurrencyAmount function.

Reply:
is the purpose to avoid a new function in CurrencyAmount class?

can i look at it later, but before this PR merged into master?

Reply:
yes. and to make the logic clear. ofc, no problem.

Reply:
CurrencyAmount.fromCurrencyAmount logic is very clear to me

Reply:
done